bot/exts/utilities/bookmark.py

```diff
@@ -16,6 +16,13 @@
 # Number of seconds to wait for other users to bookmark the same message
 TIMEOUT = 120
 BOOKMARK_EMOJI = "📌"
+MESSAGE_NOT_FOUND_ERROR = (
+    "You must either provide a reference to a valid message, or reply to one."
+    "\n\nThe lookup strategy for a message is as follows (in order):"
+    "\n1. Lookup by '{channel ID}-{message ID}' (retrieved by shift-clicking on 'Copy ID')"
+    "\n2. Lookup by message ID (the message **must** be in the current channel)"
```

Comment:
This lookup fails for any non-cached message for me. Looking at the source code for the converter, after a cache-miss, it'll try to get the channel, then fetch the message from it, but it always fails to get the channel, even when it has already done other operations in that channel (such as sending a bookmark or a failure).

The reason I commented on this portion specifically is that I think it's best to just not mention this, and not worry about it.

Reply:
This is fixed in a more recent version of d.py https://github.com/Rapptz/discord.py/pull/7688

Until we bump up to that, this monkey patch is supposed to resolve the issue you are seeing.
https://github.com/python-discord/sir-lancebot/blob/main/bot/monkey_patches.py#L72

Reply:
Testing it myself, it seems like this message converter just does not have a good time with messages in DM channels for whatever reason.

Possibly we just make it only work with a reply, since that seems to work consistently

Reply:
I think that’s fine, you’re going to be at the message when running the command anyway

Reply:
[python-discord/sir-lancebot@`36eb3d1` (#1049)](https://github.com/python-discord/sir-lancebot/pull/1049/commits/36eb3d19c742dac3351ff90e44509115867a512b)